### PR TITLE
Do not cancel classification change notifications

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Now that we're reporting minimal changes, it's important that all change notifications go through.  So this part of classification is no longer cancellable on normal edits.

@jasonmalinowski  and i also have a plan on how to greatly simplify all this code as we think this is all far too complex for no gain.

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs [hunk 1]
```diff
@@ -35,12 +35,6 @@ internal partial class SyntacticClassificationTaggerProvider
         /// </summary>
         internal partial class TagComputer
         {
-            // This is how long we will wait after completing a parse before we tell the editor to
-            // re-classify the file. We do this, since the edit may have non-local changes, or the user
-            // may have made a change that introduced text that we didn't classify because we hadn't
-            // parsed it yet, and we want to get back to a known state.
-            private const int ReportChangeDelayInMilliseconds = TaggerConstants.ShortDelay;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs:42]: from my convo with Jason, this field becamse useless.  we waited a bit to give our bg system time to parse the doc.  but we're always parsing the doc prior to reporting the change (in order to determine what changed).  so any amount of extra delay is superfluous.

File: src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs [hunk 2]
```diff
@@ -235,14 +226,12 @@ private async Task EnqueueProcessSnapshotWorkerAsync(Document currentDocument, C
...
 
-                _reportChangeCancellationSource = new CancellationTokenSource();
                 _notificationService.RegisterNotification(() =>
                     {
                         _workQueue.AssertIsForeground();
                         ReportChangedSpan(changedSpan);
                     },
-                    ReportChangeDelayInMilliseconds,
-                    _listener.BeginAsyncOperation("ReportEntireFileChanged"),
+                    _listener.BeginAsyncOperation("EnqueueProcessSnapshotWorkerAsync.ReportChangedSpan"),
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs:234]: Use nameof?

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs [hunk 3]
```diff
@@ -156,9 +156,10 @@ public bool SupportsSemanticModel
         /// </summary>
         /// <returns>
         /// The returned syntax tree can be <see langword="null"/> if the <see cref="SupportsSyntaxTree"/> returns <see
-        /// langword="false"/>. This function will return the same value if called multiple times. Specifically, the
-        /// same Task instance will be returned, which will always return the same underlying <see cref="SyntaxTree"/>
-        /// instance.
+        /// langword="false"/>. This function may cause computation to occurr the first time it is called, but will return
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs:159]:
```suggestion
        /// langword="false"/>. This function may cause computation to occur the first time it is called, but will return
```

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs [hunk 4]
```diff
@@ -156,9 +156,10 @@ public bool SupportsSemanticModel
...
+        /// a cached result every subsequent time.  <see cref="SyntaxTree"/>'s can hold onto their roots lazily. So calls
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs:160]:
```suggestion
        /// a cached result every subsequent time.  <see cref="SyntaxTree"/>s can hold onto their roots lazily. So calls 
```

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**